File: fixtures/scale/lua/medium/src/mathutil.lua
local function guard_number(x)
  local value = tonumber(x)
  if value ~= value then error("not a number") end
  return value
end

local function guard_positive(x)
  local value = guard_number(x)
  if value <= 0 then error("must be positive") end
  return value
end

return { guard_number = guard_number, guard_positive = guard_positive }

File: fixtures/scale/javascript/medium/src/mathutil.js
function guardNumber(x) {
  const value = Number(x);
  if (Number.isNaN(value)) throw new Error('not a number');
  return value;
}

function guardPositive(x) {
  const value = guardNumber(x);
  if (value <= 0) throw new Error('must be positive');
  return value;
}

export default { guardNumber, guardPositive };
